### PR TITLE
[PowerPC][JITLink] Support R_PPC64_GOT_PCREL34

### PR DESCRIPTION
**Pull request by bzEq**
`R_PPC64_GOT_PCREL34` is generated for pwr10+.

### CODE REVIEW

**Review by vgvassilev — APPROVED** [review]
This looks good to me and our use case in https://github.com/cms-sw/cmssw/issues/41658. I'd wait for one more reviewer to approve it. @lhames ping.

**Review by lhames — APPROVED** [review]
Looks good to me. Thanks @bzEq!

**Pull request merged**
